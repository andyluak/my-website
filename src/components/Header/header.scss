.header {
	padding: 1em 6em;
	display:flex;
	justify-content: space-between;
	align-items: center;
	background: white;
}

nav ul {
	display: flex;
	align-items: center;

	li {
		padding: 0 2em;
		cursor: pointer;
		font-weight: 300;
	}

	li:hover {
		font-weight: 500;
	}
}

@media screen and (max-width: 768px) {
	.header {
		padding: .3em 1em;

		.logo {
			width: 100px;
		}
	}

	nav ul {
		align-items: flex-start;

		li {
			font-size: .8em;
			padding: 0 .4em;
		}
	}
}