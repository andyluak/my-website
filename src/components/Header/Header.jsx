import './header.scss';

import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { ReactComponent as Logo } from '../../assets/images/logo.svg';

const Header = ({isAuthenticated}) => {
	return <div className="header">
		<Logo className="logo" />
		<nav>
			<ul>
				<li><Link to="/">Home</Link> </li>
				<li><Link to="/blog">Blog </Link> </li>
				{ isAuthenticated && <li><Link to="/admin">Admin</Link> </li> }
			</ul>
		</nav>
	</div>;
};
const mapStateToProps = state => ({
	isAuthenticated: state.auth.isAuthenticated,
});

export default connect(mapStateToProps)(Header);
