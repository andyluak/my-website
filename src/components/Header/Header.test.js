import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe( 'Header component tests', () =>{
	describe( 'Header not authenticated component tests', () => {
		let wrapper;
		beforeEach( () => {
			const mockProps = {
				isAuthenticated: false,
			}
			wrapper = shallow( <Header {...mockProps} /> );
		});
		it( 'expect Header to render without crashing', () => {
			expect(wrapper).toMatchSnapshot();
		})

		it( 'expect to have 2 links in the header', () => {
			expect(wrapper.find('Link')).toHaveLength(2);
		});


	})


	describe( 'Header authorized component tests', () => {
		let wrapper;
		beforeEach( () => {
			const mockProps = {
				isAuthenticated: true,
			}
			wrapper = shallow( <Header {...mockProps} /> );
		});

		it( 'expect to have 3 links in the header', () => {
			expect(wrapper.find('Link')).toHaveLength(3);
		})
	})
})


