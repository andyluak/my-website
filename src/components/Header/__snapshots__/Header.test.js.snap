// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests Header not authenticated component tests expect Header to render without crashing 1`] = `
<div
  className="header"
>
  <ForwardRef(SvgLogo)
    className="logo"
  />
  <nav>
    <ul>
      <li>
        <Link
          to="/"
        >
          Home
        </Link>
         
      </li>
      <li>
        <Link
          to="/blog"
        >
          Blog 
        </Link>
         
      </li>
    </ul>
  </nav>
</div>
`;
